var/spack/repos/builtin/packages/hepmc3/package.py

```diff
@@ -42,8 +43,6 @@ class Hepmc3(CMakePackage):
     def cmake_args(self):
         spec = self.spec
         args = [
-            '-Dmomentum:STRING=GEV',
-            '-Dlength:STRING=MM',
```

Comment:
I assume these are needed to install the older versions? You should move the definitions to an `if spec.satisfies('@:3.2.3'):`

Reply:
No, those apply to hepmc 2, which has its own recipe.